app/controllers/account_controller.rb

```diff
@@ -424,13 +424,14 @@ def successful_authentication(user, reset_stages: true)
   end
 
   def login_user!(user)
-    # Valid user
-    self.logged_user = user
     # generate a key and set cookie if autologin
-    if params[:autologin] && Setting.autologin?
+    if Setting.autologin? && (params[:autologin] || session[:autologin_requested])
       set_autologin_cookie(user)
```

Comment:
The `autologin_requested` value can be deleted after the cookie has been set, right?